sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md

```diff
@@ -123,7 +123,7 @@ export interface SecretProperties {
     enabled?: boolean;
     readonly expiresOn?: Date;
     id?: string;
-    readonly keyId?: URL;
+    readonly keyId?: string;
```

Comment:
This would be a breaking change for someone who has the below code
```js
if (mySecret.keyId) {
     console.log(mySecret.keyId.toJSON());
}
```

One approach could be that we deprecate this property and use a new name, perhaps something to denote that this is related to the KV certificate?

cc @bterlson, @xirzec

Reply:
KeyVault Certificates uses string already. That `if` that you mention couldn't have been used because this property was never received as `keyId`.

Reply:
If someone has written the code Ramya wrote above with a KeyVaultSecret, then it will break their compilation if we change the type signature to string, even though the current implementation can't set it to anything at runtime.

Reply:
What is the recommended path forward? If it's a version change, I can do that.

Reply:
This property was introduced in 4.0 GA, so even though 4.1 is in preview, it create a breaking change between 4.0 and 4.1 to remove something from the API.

I think we have to put a deprecation on the docs for `keyId` that mentions that the property will never be defined and create a new property name (I agree with Ramya that something that lets a customer know that this is about correlation to a certificate would be good). When we eventually do 5.0 we can remove the property entirely.

Reply:
I don't think we should deprecate `keyId` since it will be a misalignment with the other languages.

Reply:
we could deprecate keyId in the GA versions and re-introduce it in 4.1.0? in the GA versions we would introduce `kid`, which is the one that gets populated? (then deprecate `kid` again in 4.1.0)